tuf_conformance/repository_simulator.py

```diff
@@ -155,11 +155,12 @@ def any_targets(self, role: str) -> Targets:
         assert isinstance(signed, Targets)
         return signed
 
-    def all_targets(self) -> Iterator[tuple[str, Targets]]:
+    def all_signed_targets(self) -> Iterator[tuple[str, Targets]]:
         """Yield role name and signed portion of targets one by one."""
-        for role, md in self.mds.items():
-            if role not in [Root.type, Timestamp.type, Snapshot.type]:
-                yield role, md.signed
+        for role, mds in self.signed_mds.items():
+            for md in mds:
+                if role not in [Root.type, Timestamp.type, Snapshot.type]:
+                    yield role, Metadata.from_bytes(md).signed
```

Comment:
hmm, rereading this, it now yields _every version_ of every targets metadata. I think what we want is _current version_ of every targets metadata

This probably already has correct test results since the only call site will end up rewriting the same variables for each version so always uses the current version last...  so we could leave it as is and file an issue to refactor the whole method away later (by storing hashes/length when targets roles are published).

But removing the inner loop is fine too

Reply:
Let's revisit this once we get to the tests after the repo is set up nicely. Keeping it open.